pmlc/ast/builder.cc

```diff
@@ -752,6 +768,106 @@ struct ProgramBuilder {
         .result();
   }
 
+  Value handleLoop(ExprNodeLoop *node) {
+    // take lowBound, upperBound, step from operands.
+    auto indexs = node->indexs;
+    if (indexs.size() != 3) {
+      throw std::runtime_error(
+          "please check the index of loop, lbound, ubound and step");
+    }
+    auto indexType = builder.getIndexType();
+    std::vector<Value> indexValue;
+    for (auto index : indexs) {
+      auto indexNum = builder.create<mlir::IndexCastOp>(
+          loc, builder.lookupNode(index), indexType);
+      indexValue.push_back(indexNum);
+    }
+
+    auto operands = node->operands;
+    auto results = node->results;
+    if (operands.size() != results.size()) {
+      throw std::runtime_error(
+          "In scfFor op, init Iter number have to equal Iter args");
```

Comment:
Same as above.

Reply:
for this check, I think I already check it in edsl level. like below:
```
    if (iterTensor.size() != yieldTensor.size()) {
      throw ffi_exception("iter args don't equal to init args of scf", edsl_source_location::current());
    } 
```
maybe I should remove this one also.